lib/index.js

```diff
@@ -411,9 +382,68 @@ var StateMachine = stampit({
       this.addState(event.to);
     }, this);
 
+    this.mixin = _(this.events)
+      .mapValues(function (event, name) {
+        return this.buildEvent(name);
+      }, this)
+      .assign({
+        can: this.can.bind(this),
+        cannot: this.cannot.bind(this),
+        is: this.is.bind(this),
+        hasState: this.hasState.bind(this),
+        isFinal: this.isFinal.bind(this)
+      })
+      .value();
+
     this.current = this.initial;
-    return this.initTarget(_.first(context.args));
   }
 });
 
-module.exports = StateMachine;
+var Target = stampit({
+  props: {
+    events: {},
+    callbacks: {},
+    initial: 'none',
+    final: null
+  },
+  static: {
+    StateMachine: StateMachine,
+    Promise: global.Promise || require('es6-promise').Promise,
+    FsmError: FsmError
+  },
+  init: function init (context) {
+    var target = _.first(context.args) || this;
+
+    _.defaults(target, {
+      events: this.events,
+      callbacks: this.callbacks,
+      initial: this.initial,
+      final: this.final
+    }, {
+      events: {},
+      callbacks: {},
+      initial: 'none',
+      final: null
+    });
+
+    var stateMachine = context.stamp.StateMachine({
+      target: target,
+      events: target.events,
+      callbacks: target.callbacks,
+      initial: target.initial,
+      final: target.final,
+      Promise: context.stamp.Promise,
```

Comment:
this could be more elegant, I guess.  I'm not sure why `FsmError` is exposed except for testing.

an idea about choosing the `Promise` implementation is to provide a static `usePromise()` function on `Target`:

```js
var Target = stampit({
  // stuff
  static: {
    usePromise (Promise) {
      this.fixed.static.StateMachine.Promise = Promise;
    }
  }
});
```

usage:

```js
var StateMachine = require('fsm-as-promised');
StateMachine.usePromise(require('q'));
```